ArtOfIllusion/src/artofillusion/Scene.java

```diff
@@ -810,32 +794,23 @@ public void replaceImage(int which, ImageMap im)
 
   public void replaceObject(Object3D original, Object3D replaceWith, UndoRecord undo)
   {
-    for (int i = 0; i < objects.size(); i++)
-      {
-        ObjectInfo info = objects.elementAt(i);
-        if (info.getObject() != original)
-          continue;
-        if (undo != null)
-          undo.addCommand(UndoRecord.SET_OBJECT, info, original);
-        info.setObject(replaceWith);
-        info.clearCachedMeshes();
-      }
+    Optional<UndoRecord> optionalUndo = Optional.ofNullable(undo);
+    objects.stream().filter(item -> item.getObject() == original).forEach(item -> {
+      optionalUndo.ifPresent(command -> command.addCommand(UndoRecord.SET_OBJECT, item, original));
+      item.setObject(replaceWith);
+      item.clearCachedMeshes();
+    });
   }
 
   /** This should be called whenever an object changes.  It clears any cached meshes for
       any instances of the object. */
 
   public void objectModified(Object3D obj)
   {
-    for (int i = 0; i < objects.size(); i++)
-      {
-        ObjectInfo info = objects.elementAt(i);
-        if (info.getObject() == obj)
-          {
-            info.clearCachedMeshes();
-            info.setPose(null);
-          }
-      }
+    objects.stream().filter(item -> item.getObject() == obj).forEach(item -> {
```

Comment:
And this as well, though the `.filter` clause can probably stay on the first line here.